File: token.py
from enum import Enum, auto

class TokenType(Enum):
    # single-character tokens
    LEFT_PAREN = auto(); RIGHT_PAREN = auto()
    LEFT_BRACE = auto(); RIGHT_BRACE = auto()
    COMMA = auto(); DOT = auto(); MINUS = auto(); PLUS = auto()
    SEMICOLON = auto(); SLASH = auto(); STAR = auto()
    EQUAL = auto(); LESS = auto(); GREATER = auto(); BANG = auto()

    # multi-character operators
    EQUAL_EQUAL = auto()
    BANG_EQUAL = auto()
    LESS_EQUAL = auto()
    GREATER_EQUAL = auto()

    # literals
    STRING = auto()
    NUMBER = auto()

    EOF = auto()


class Token:
    def __init__(self, type_, lexeme, literal, line):
        self.type = type_
        self.lexeme = lexeme
        self.literal = literal
        self.line = line

    def __repr__(self):
        return f"Token(type={self.type}, lexeme='{self.lexeme}', literal={self.literal}, line={self.line})"

File: Token.kt
/*
 * LAB 5
 * Functions
 */

enum class TokenType {
    // Structure
    LEFT_PAREN, RIGHT_PAREN,
    LEFT_BRACE, RIGHT_BRACE,
    SEMICOLON, COMMA,

    // Gen Z verbs (expressions)
    MIX, TAKE_AWAY, MULTIPLY, DIVIDE, FLIP, CHECK_IF,

    // Statement keywords (existing + new cookbook)
    REMEMBER, SET, PRINT, AS, TO,
    THE, VALUE, WHEN, YOU,

    // New cookbook-style control flow keywords
    IF, ELSE,           // mapped from "when" / "otherwise"
    WHILE, FOR,         // "repeat" / "for"
    FUN, RETURN,        // "recipe" / "serve"

    // Connectors for expressions
    AND, FROM, WITH, OR,

    // Comparison
    GREATER, LESS, EQUAL_EQUAL,

    // Literals / identifiers
    IDENTIFIER,
    STRING,
    NUMBER,
    TRUE,
    FALSE,

    EOF
}

data class Token(
    val type: TokenType,
    val lexeme: String,
    val literal: Any?,
    val line: Int
)
